lms/services/launch_verifier.py

```diff
@@ -50,44 +49,78 @@ def verify(self):
             raise self._exception
 
     def _verify(self):
-        try:
-            consumer_key = self._request.params["oauth_consumer_key"]
-        except KeyError as err:
-            raise NoConsumerKey() from err
+        is_valid, _request = self._oauth1_endpoint.validate_request(
+            uri=self._request.url,
+            http_method=self._request.method,
+            body=self._request.body,
+            headers=dict(self._request.headers),
```

Comment:
The behavior of this method is quite different from the library we used before and is the source of all the change in the tests. Previously this accepted `uri`, `params` and `body`, now it only accepts `uri` and `body` and the params are expected to be in the `uri`.

This causes quite a lot of annoyance, as our tests relied on being able to modify the params separately quite a bit. More on this later...